checks.d/mongo.py

```diff
@@ -730,3 +733,41 @@ def check(self, instance):
                         submit_method(self, metric_name_alias, value, tags=ns_tags)
             except Exception, e:
                 self.log.warning('Failed to record `top` metrics %s' % str(e))
+
+        # Fetch information analogous to Mongo's db.getReplicationInfo()
+        localdb = cli['local']
+
+        oplog_data = {}
+
+        for ol_collection_name in ("oplog.rs", "oplog.$main", None):
+            ol_metadata = localdb.system.namespaces.find_one({"name": "local.%s" % ol_collection_name})
+            if ol_metadata is not None:
+                break
+
+        if ol_collection_name is not None:
+            try:
+                oplog_data['totalSizeMB'] = round(ol_metadata['options']['size'] / 2.0 ** 20, 2)
+
+                oplog = localdb[ol_collection_name]
+
+                oplog_data['usedSizeMB'] = round(localdb.command("collstats", ol_collection_name)['size'] / 2.0 ** 20, 2)
+
+                op_asc_cursor = oplog.find().sort("_id", pymongo.ASCENDING).limit(1)
+                op_dsc_cursor = oplog.find().sort("_id", pymongo.DESCENDING).limit(1)
+
+                try:
+                    first_timestamp = op_asc_cursor[0]['ts'].as_datetime()
+                    last_timestamp = op_dsc_cursor[0]['ts'].as_datetime()
+                    oplog_data['timeDiffSeconds'] = (last_timestamp - first_timestamp).total_seconds()
```

Comment:
Would it make sense to you to break that out into a function of its own, e.g.

```
        def total_seconds(td):
            """
            Returns total seconds of a timedelta in a way that's safe for
            Python < 2.7
            """
            if hasattr(td, 'total_seconds'):
                return td.total_seconds()
            else:
                return (
                    lag.microseconds +
                    (lag.seconds + lag.days * 24 * 3600) * 10**6
                ) / 10.0**6
```
and then just call `data['replicationLag'] = total_seconds(lag)` and `oplog_data['timeDiffSeconds'] = total_seconds(last_timestamp - first_timestamp)`? Or would it be easier if I just reduplicated the `total_seconds` check from the `replicationLag` part?